pkg/ring/kv/metrics.go

```diff
@@ -18,6 +20,17 @@ func init() {
 	requestDuration.Register()
 }
 
+// errorCode converts an error into an HTTP status code, modified from weaveworks/common/instrument
+func errorCode(err error) string {
+	if err == nil {
+		return "200"
+	}
+	if resp, ok := httpgrpc.HTTPResponseFromError(err); ok {
+		return strconv.Itoa(int(resp.GetCode()))
+	}
+	return "500"
```

Comment:
This is a very strong assumption, that is not true at all when using memberlist-based client. (There are no HTTP calls at all)